packages/rspeedy/plugin-react/src/entry.ts

```diff
@@ -137,25 +139,28 @@ export function applyEntry(
         })
         // in standalone lazy bundle mode, we do not add
         // other entries to avoid wrongly exporting from other entries
-        .when(isDev && !isWeb, entry => {
-          // use prepend to make sure it does not affect the exports
-          // from the entry
-          entry
-            // This is aliased in `@lynx-js/rspeedy`
-            .prepend({
-              layer: LAYERS.BACKGROUND,
-              import: '@rspack/core/hot/dev-server',
-            })
-            .prepend({
-              layer: LAYERS.BACKGROUND,
-              import: '@lynx-js/webpack-dev-transport/client',
-            })
-            // This is aliased in `./refresh.ts`
-            .prepend({
-              layer: LAYERS.BACKGROUND,
-              import: '@lynx-js/react/refresh',
-            })
-        })
+        .when(
+          isDev && !isWeb && !(hmr === false && liveReload === false),
+          entry => {
+            // use prepend to make sure it does not affect the exports
+            // from the entry
+            entry
+              // This is aliased in `@lynx-js/rspeedy`
+              .prepend({
+                layer: LAYERS.BACKGROUND,
+                import: '@rspack/core/hot/dev-server',
+              })
+              .prepend({
+                layer: LAYERS.BACKGROUND,
+                import: '@lynx-js/webpack-dev-transport/client',
+              })
+              // This is aliased in `./refresh.ts`
+              .prepend({
+                layer: LAYERS.BACKGROUND,
+                import: '@lynx-js/react/refresh',
+              })
```

Comment:
- When enable `hmr`， only prepend `@lynx-js/react/refresh` and `@rspack/core/hot/dev-server`
- When enable `liveReload` or `hmr`, prepend `@lynx-js/webpack-dev-transport/client`